flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -892,9 +900,12 @@ private void onTriggerFailure(@Nullable PendingCheckpoint checkpoint, Throwable
                         job,
                         numUnsuccessful,
                         throwable);
-                final CheckpointException cause =
-                        getCheckpointException(
-                                CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE, throwable);
+                CheckpointFailureReason defaultReason =
+                        CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE;
+                if (throwable instanceof IOException) {
```

Comment:
The same here. Why are you so sure that IOException would not be wrapped by other exceptions?

Reply:
You are right and I have also made a change here.